sklearn/tests/test_impute.py

```diff
@@ -1147,35 +1147,48 @@ def test_missing_indicator_sparse_no_explicit_zeros():
 
 
 @pytest.mark.parametrize("marker", [np.nan, -1, 0])
-def test_imputation_add_indicator(marker):
+@pytest.mark.parametrize("imputer_constructor",
+                         [SimpleImputer, IterativeImputer])
+def test_imputers_add_indicator(marker, imputer_constructor):
```

Comment:
Since this test is for testing the indicator, you can simplify it by only looking at the last 4 columns for `X_trans` (and keep the same `X` you had originally).

```python
X = np.array([
    [marker, 1,      5,       marker, 1],
    [2,      marker, 1,       marker, 2],
    [6,      3,      marker,  marker, 3],
    [1,      2,      9,       marker, 4]
])
X_true_indicator = np.array([
    [1., 0., 0., 1.],
    [0., 1., 0., 1.],
    [0., 0., 1., 1.],
    [0., 0., 0., 1.]
])
imputer = imputer_constructor(missing_values=marker,
                                add_indicator=True)

X_trans = imputer.fit(X).transform(X)
assert_allclose(X_trans[:, -4:], X_true_indicator)
```

Reply:
👍